### PR TITLE
Add recipe for egobox

### PR DESCRIPTION
**Pull request by relf**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by relf** [comment]
Hi @conda-forge/help-python, ready for review, thanks!

**Review by synapticarbors — CHANGES_REQUESTED** [review]
Please see the following about adding bundled licenses for rust:
https://conda-forge.org/docs/maintainer/adding_pkgs/#rust

**Comment by relf** [comment]
@synapticarbors thanks for reviewing!

**Comment by relf** [comment]
@conda-forge/help-rust, the PR looks good to me, anything else to do?

File: recipes/egobox/conda_build_config.yaml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+MACOSX_DEPLOYMENT_TARGET: # [osx and x86]
```
Comment (xhochy) [inline recipes/egobox/conda_build_config.yaml:1]:
```suggestion
c_stdlib_version: # [osx and x86]
```

File: recipes/egobox/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+build:
+  skip: true  # [py<38 or python_impl == 'pypy']
+  script: 
+    - cargo-bundle-licenses --format yaml --output THIRDPARTY.yml
+    - {{ PYTHON }} -m pip install . -vv
+  number: 0
+
+requirements:
+  build:
+    - {{ compiler('c') }}
```
Comment (xhochy) [inline recipes/egobox/meta.yaml:21]:
```suggestion
    - {{ stdlib('c') }}
    - {{ compiler('c') }}
```

File: recipes/egobox/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    - {{ compiler('rust') }}
+    - cargo-bundle-licenses
+  host:
+    - python
+    - maturin >=1.4,<2.0
+    - pip
+  run:
+    - python
+    - __osx >={{ MACOSX_DEPLOYMENT_TARGET|default("10.9") }}  # [osx and x86_64]
```
Comment (xhochy) [inline recipes/egobox/meta.yaml:30]:
```suggestion
```

**Review by xhochy — CHANGES_REQUESTED** [review]
See https://conda-forge.org/news/2024/03/24/stdlib-migration/

**Comment by relf** [comment]
@xhochy, thanks for reviewing.

**Review by xhochy — APPROVED** [review]

**Pull request merged**